diff --git nethack-3.4.3.b/dat/Makefile nethack-3.4.3/dat/Makefile
index 75a5aa8..2b69c06 100644
--- nethack-3.4.3.b/dat/Makefile
+++ nethack-3.4.3/dat/Makefile
@@ -101,7 +101,7 @@ options:	../util/makedefs
 
 spec_levs: ../util/lev_comp \
 	bigroom.des castle.des endgame.des gehennom.des knox.des medusa.des \
-	mines.des oracle.des sokoban.des tower.des yendor.des
+	mines.des oracle.des ring.des sokoban.des tower.des yendor.des
 	../util/lev_comp bigroom.des
 	../util/lev_comp castle.des
 	../util/lev_comp endgame.des
@@ -110,6 +110,7 @@ spec_levs: ../util/lev_comp \
 	../util/lev_comp medusa.des
 	../util/lev_comp mines.des
 	../util/lev_comp oracle.des
+	../util/lev_comp ring.des
 	../util/lev_comp sokoban.des
 	../util/lev_comp tower.des
 	../util/lev_comp yendor.des
diff --git nethack-3.4.3.b/dat/dungeon.def nethack-3.4.3/dat/dungeon.def
index fbb17bc..a142e7f 100644
--- nethack-3.4.3.b/dat/dungeon.def
+++ nethack-3.4.3/dat/dungeon.def
@@ -18,6 +18,7 @@ DUNGEON:	"The Dungeons of Doom" "D" (25, 5)
 ALIGNMENT:	unaligned
 BRANCH:		"The Gnomish Mines" @ (2, 3)
 %REINCARNATION LEVEL:		"rogue" "R" @ (15, 4)
+LEVEL:		"ring" "none" @ (1, 0)
 LEVEL:		"oracle" "O" @ (5, 5)
 LEVALIGN:	neutral
 CHAINBRANCH:	"Sokoban" "oracle" + (1, 0) up
diff --git nethack-3.4.3.b/util/lev_main.c nethack-3.4.3/util/lev_main.c
index bff4668..b4b5fdc 100644
--- nethack-3.4.3.b/util/lev_main.c
+++ nethack-3.4.3/util/lev_main.c
@@ -241,6 +241,7 @@ char **argv;
 				":dat:medusa.des",
 				":dat:mines.des",
 				":dat:oracle.des",
+				":dat:ring.des",
 				":dat:sokoban.des",
 				":dat:tower.des",
 				":dat:yendor.des"
diff --git nethack-3.4.3.b/win/win32/levcomp.dsp nethack-3.4.3/win/win32/levcomp.dsp
index 7c1486e..e1a4de3 100644
--- nethack-3.4.3.b/win/win32/levcomp.dsp
+++ nethack-3.4.3/win/win32/levcomp.dsp
@@ -71,6 +71,7 @@ echo monk.des	..\util\levcomp.exe monk.des	 \
 echo oracle.des	..\util\levcomp.exe oracle.des	 \
 echo priest.des	..\util\levcomp.exe priest.des	 \
 echo ranger.des	..\util\levcomp.exe ranger.des	 \
+echo ring.des	..\util\levcomp.exe ring.des	 \
 echo rogue.des	..\util\levcomp.exe rogue.des	 \
 echo samurai.des	..\util\levcomp.exe samurai.des	 \
 echo sokoban.des	..\util\levcomp.exe sokoban.des	 \
@@ -125,6 +126,7 @@ echo monk.des	..\util\levcomp.exe monk.des	 \
 echo oracle.des	..\util\levcomp.exe oracle.des	 \
 echo priest.des	..\util\levcomp.exe priest.des	 \
 echo ranger.des	..\util\levcomp.exe ranger.des	 \
+echo ring.des	..\util\levcomp.exe ring.des	 \
 echo rogue.des	..\util\levcomp.exe rogue.des	 \
 echo samurai.des	..\util\levcomp.exe samurai.des	 \
 echo sokoban.des	..\util\levcomp.exe sokoban.des	 \
