diff -Naur nethack-3.4.3.b/src/Makefile nethack-3.4.3/src/Makefile
--- nethack-3.4.3.b/src/Makefile	2013-01-29 20:01:15.982002656 +0100
+++ nethack-3.4.3/src/Makefile	2013-01-29 20:29:23.259528763 +0100
@@ -220,7 +220,9 @@
 #
 #
 WINSRC = $(WINTTYSRC)
+WINSRC += $(PFASRC)/middleman.c
 WINOBJ = $(WINTTYOBJ)
+WINOBJ += middleman.o
 
 # on some systems the termcap library is in -ltermcap or -lcurses
 # on 386 Xenix, the -ltermlib tputs() seems not to work; use -lcurses instead
@@ -610,6 +612,8 @@
 wintty.o: ../win/tty/wintty.c $(HACK_H) ../include/dlb.h \
 		../include/patchlevel.h ../include/tcap.h
 	$(CC) $(CFLAGS) -c ../win/tty/wintty.c
+middleman.o: $(PFASRC)/middleman.c $(PFAINC)/middleman.h
+	$(CC) $(CFLAGS) -c $(PFASRC)/middleman.c
 Window.o: ../win/X11/Window.c ../include/xwindowp.h ../include/xwindow.h \
 		$(CONFIG_H)
 	$(CC) $(CFLAGS) -c ../win/X11/Window.c
--- nethack-3.4.3.b/src/windows.c	2003-12-08 00:39:13.000000000 +0100
+++ nethack-3.4.3/src/windows.c	2013-01-29 20:30:53.579039253 +0100
@@ -3,6 +3,7 @@
 /* NetHack may be freely redistributed.  See license for details. */
 
 #include "hack.h"
+#include "middleman.h"
 #ifdef TTY_GRAPHICS
 #include "wintty.h"
 #endif
@@ -100,7 +101,9 @@
 
     for(i=0; winchoices[i].procs; i++)
 	if (!strcmpi(s, winchoices[i].procs->name)) {
-	    windowprocs = *winchoices[i].procs;
+		windowprocs = middleman;
+		real_winprocs = *winchoices[i].procs;
+	    //windowprocs = *winchoices[i].procs;
 	    if (winchoices[i].ini_routine) (*winchoices[i].ini_routine)();
 	    return;
 	}
