diff -Naur nethack-3.4.3.b/src/Makefile nethack-3.4.3/src/Makefile
--- nethack-3.4.3.b/src/Makefile	2013-01-06 19:41:52.482424960 +0100
+++ nethack-3.4.3/src/Makefile	2013-01-06 19:43:10.342012210 +0100
@@ -141,6 +141,9 @@
 #LINK = gcc
 #LFLAGS = -Xlinker -soname=_APP_
 
+# PFA
+PFASRC = ../../pfa-src
+
 # Only used for the Gnome interface.
 # When including the Gnome interface, you need to include gnome specific
 # directories.  The ones given below is the usual spot for linux systems.
@@ -151,7 +154,7 @@
 # flags for debugging:
 # CFLAGS = -g -I../include
 
-CFLAGS = -O -I../include
+CFLAGS = -O -I../include -I$(PFASRC)
 LFLAGS = 
 
 # The Qt and Be window systems are written in C++, while the rest of
@@ -214,8 +217,8 @@
 
 #
 #
-WINSRC = $(WINTTYSRC)
-WINOBJ = $(WINTTYOBJ)
+WINSRC = $(WINTTYSRC) $(PFASRC)/middleman.c
+WINOBJ = $(WINTTYOBJ) middleman.o
 
 # on some systems the termcap library is in -ltermcap or -lcurses
 # on 386 Xenix, the -ltermlib tputs() seems not to work; use -lcurses instead
@@ -571,6 +574,9 @@
 		../include/wintty.h ../include/trampoli.h
 	touch $(HACK_H)
 #
+middleman.o: $(PFASRC)/middleman.c $(PFASRC)/middleman.h
+	$(CC) $(CFLAGS) -c $(PFASRC)/middleman.c
+#
 tos.o: ../sys/atari/tos.c $(HACK_H) ../include/tcap.h
 	$(CC) $(CFLAGS) -c ../sys/atari/tos.c
 pcmain.o: ../sys/share/pcmain.c $(HACK_H) ../include/dlb.h \
diff -Naur nethack-3.4.3.b/src/windows.c nethack-3.4.3/src/windows.c
--- nethack-3.4.3.b/src/windows.c	2013-01-06 19:41:52.492424908 +0100
+++ nethack-3.4.3/src/windows.c	2013-01-06 19:43:24.228605240 +0100
@@ -3,6 +3,8 @@
 /* NetHack may be freely redistributed.  See license for details. */
 
 #include "hack.h"
+#include "middleman.h"
+
 #ifdef TTY_GRAPHICS
 #include "wintty.h"
 #endif
@@ -100,7 +102,9 @@
 
     for(i=0; winchoices[i].procs; i++)
 	if (!strcmpi(s, winchoices[i].procs->name)) {
-	    windowprocs = *winchoices[i].procs;
+		windowprocs = middleman;
+		real_winprocs = *winchoices[i].procs;
+	    //windowprocs = *winchoices[i].procs;
 	    if (winchoices[i].ini_routine) (*winchoices[i].ini_routine)();
 	    return;
 	}
