diff --git a/nethack-3.4.3/src/lock.c b/nethack-3.4.3/src/lock.c
index 4d5d333..00184c7 100644
--- a/nethack-3.4.3/src/lock.c
+++ b/nethack-3.4.3/src/lock.c
@@ -566,7 +566,7 @@ doopen()		/* try to open a door */
 		door->doormask = D_NODOOR;
 		if (*in_rooms(cc.x, cc.y, SHOPBASE)) add_damage(cc.x, cc.y, 0L);
 	    } else
-		door->doormask = D_ISOPEN;
+		door->doormask = D_NODOOR;
 	    if (Blind)
 		feel_location(cc.x,cc.y);	/* the hero knows she opened it  */
 	    else
