--- nethack-3.4.3.b/src/detect.c
+++ nethack-3.4.3/src/detect.c
@@ -921,6 +921,7 @@ register int x, y;
     /* Secret corridors are found, but not secret doors. */
     if (lev->typ == SCORR) {
 	lev->typ = CORR;
+	pfa_scorr_discovery(x, y);
 	unblock_point(x,y);
     }
 
@@ -995,6 +996,7 @@ struct rm *lev;
 
 	lev->typ = DOOR;
 	lev->doormask = newmask;
+	pfa_sdoor_discovery(-1, -1);//location unknown
 }
 
 
@@ -1013,6 +1015,7 @@ genericptr_t num;
 		(*(int*)num)++;
 	} else if(levl[zx][zy].typ == SCORR) {
 		levl[zx][zy].typ = CORR;
+		pfa_scorr_discovery(zx, zy);
 		unblock_point(zx,zy);
 		magic_map_background(zx, zy, 0);
 		newsym(zx, zy);
@@ -1080,6 +1083,7 @@ genericptr_t num;
 		newsym(zx, zy);
 		(*(int*)num)++;
 	} else if(levl[zx][zy].typ == SCORR) {
+		pfa_scorr_discovery(zx, zy);
 		levl[zx][zy].typ = CORR;
 		unblock_point(zx, zy);
 		newsym(zx, zy);
@@ -1199,6 +1203,7 @@ register int aflag;
 		    } else if(levl[x][y].typ == SCORR) {
 			if(rnl(7-fund)) continue;
 			levl[x][y].typ = CORR;
+			pfa_scorr_discovery(x,y);
 			unblock_point(x,y);	/* vision */
 			exercise(A_WIS, TRUE);
 			nomul(0);
--- nethack-3.4.3.b/src/do.c
+++ nethack-3.4.3/src/do.c
@@ -7,6 +7,8 @@
 #include "hack.h"
 #include "lev.h"
 
+#include "pfamain.h"
+
 #ifdef SINKS
 # ifdef OVLB
 STATIC_DCL void FDECL(trycall, (struct obj *));
@@ -1097,6 +1099,7 @@ boolean at_stairs, falling, portal;
 		}
 		mklev();
 		new = TRUE;	/* made the level */
+		pfa_new_level_reached();
 	} else {
 		/* returning to previously visited level; reload it */
 		fd = open_levelfile(new_ledger, whynot);
