diff --git nethack-3.4.3.b/src/options.c nethack-3.4.3/src/options.c
index dfb8006..a2b97f1 100644
--- nethack-3.4.3.b/src/options.c
+++ nethack-3.4.3/src/options.c
@@ -14,6 +14,7 @@ NEARDATA struct instance_flags iflags;	/* provide linkage */
 #include "tcap.h"
 #include <ctype.h>
 #endif
+#include "pfamain.h"
 
 #define WINTYPELEN 16
 
@@ -494,7 +495,7 @@ initoptions()
 	int i;
 
 	/* initialize the random number generator */
-	setrandom();
+	pfa_setrandom();
 
 	/* for detection of configfile options specified multiple times */
 	iflags.opt_booldup = iflags.opt_compdup = (int *)0;
