diff -Naur nethack-3.4.3.b/src/Makefile nethack-3.4.3/src/Makefile
--- nethack-3.4.3.b/src/Makefile	2013-01-29 11:46:09.193483352 +0100
+++ nethack-3.4.3/src/Makefile	2013-01-29 11:57:56.319652521 +0100
@@ -148,10 +148,15 @@
 #
 GNOMEINC=-I/usr/lib/glib/include -I/usr/lib/gnome-libs/include -I../win/gnome
 
+# PFA
+PFACSRC = ../../src
+PFACINC = ../../include
+
 # flags for debugging:
 # CFLAGS = -g -I../include
 
 CFLAGS = -O -I../include
+CFLAGS += -I$(PFACINC) -L$(PFACSRC)
 LFLAGS = 
 
 # The Qt and Be window systems are written in C++, while the rest of
