diff -ur nethack-3.4.3.back/Makefile nethack-3.4.3/Makefile
--- nethack-3.4.3.back/Makefile	2012-10-27 16:03:27.179363937 +0200
+++ nethack-3.4.3/Makefile	2012-10-27 16:04:18.782424697 +0200
@@ -14,11 +14,11 @@
 # MAKE = make
 
 # make NetHack
-PREFIX	 = /usr
+PREFIX	 = ~
 GAME     = nethack
 # GAME     = nethack.prg
-GAMEUID  = games
-GAMEGRP  = bin
+GAMEUID  = `id -un`
+GAMEGRP  = `id -gn`
 
 # Permissions - some places use setgid instead of setuid, for instance
 # See also the option "SECURE" in include/config.h
@@ -35,7 +35,7 @@
 # therefore there should not be anything in GAMEDIR that you want to keep
 # (if there is, you'll have to do the installation by hand or modify the
 # instructions)
-GAMEDIR  = $(PREFIX)/games/lib/$(GAME)dir
+GAMEDIR  = $(PREFIX)/games/$(GAME)dir
 VARDIR  = $(GAMEDIR)
 SHELLDIR = $(PREFIX)/games
 
diff -ur nethack-3.4.3.back/include/unixconf.h nethack-3.4.3/include/unixconf.h
--- nethack-3.4.3.back/include/unixconf.h	2012-10-27 16:03:27.242696944 +0200
+++ nethack-3.4.3/include/unixconf.h	2012-10-27 16:04:37.075661517 +0200
@@ -37,7 +37,7 @@
 #define NETWORK		/* if running on a networked system */
 			/* e.g. Suns sharing a playground through NFS */
 /* #define SUNOS4 */	/* SunOS 4.x */
-/* #define LINUX */	/* Another Unix clone */
+#define LINUX	/* Another Unix clone */
 /* #define CYGWIN32 */	/* Unix on Win32 -- use with case sensitive defines */
 /* #define GENIX */	/* Yet Another Unix Clone */
 /* #define HISX */	/* Bull Unix for XPS Machines */
diff -ur nethack-3.4.3.back/src/Makefile nethack-3.4.3/src/Makefile
--- nethack-3.4.3.back/src/Makefile	2012-10-27 16:03:27.199363834 +0200
+++ nethack-3.4.3/src/Makefile	2012-10-27 16:05:03.215523749 +0200
@@ -230,8 +230,8 @@
 # WINTTYLIB = -ltermcap
 # WINTTYLIB = -lcurses
 # WINTTYLIB = -lcurses16
-# WINTTYLIB = -lncurses
-WINTTYLIB = -ltermlib
+WINTTYLIB = -lncurses
+# WINTTYLIB = -ltermlib
 #
 # libraries for X11
 # If USE_XPM is defined in config.h, you will also need -lXpm here.
